document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoAggregationSelectingExpressionParser.java

```diff
@@ -0,0 +1,35 @@
+package org.hypertrace.core.documentstore.mongo.parser;
+
+import java.util.Map;
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.query.Query;
+
+final class MongoAggregationSelectingExpressionParser extends MongoSelectingExpressionParser {
+
+  MongoAggregationSelectingExpressionParser(final Query query) {
+    super(query);
+  }
+
+  @Override
+  public Map<String, Object> visit(AggregateExpression expression) {
+    return new MongoAggregateExpressionParser(query).parse(expression);
+  }
+
+  @Override
+  public Object visit(final ConstantExpression expression) {
```

Comment:
Couldn't understand the reason for this change. I might need a code walkthrough with regards to these parser changes.

Reply:
Sure. I will also add some code comments to make it verbose.

In summary, we have 3 major implementations of MongoSelectingExpressionParsers:

1. MongoSelectingExpressionParser: Parses all types of expressions
2. MongoAggregationSelectingExpressionParser: Parse only AggregationExpressions (since they should go in the "$group" clause)
3. MongoNonAggregationSelectingExpressionParser: Parse all other expression types (other than AggregationExpression) (since they should go in the "$project" clause)

On top of these, we have 2 decorators, which may perform some additional operations on the parsed response.
(a). MongoIdentifierPrefixingSelectingExpressionParser: Prefixes IdentifierExpressions with "$" in front. 
(b). MongoProjectionSelectingExpressionParser: Wraps the parsed responses with supplied/generated aliases.

We use any combinations of decorators (a) and (b) with any one (1), (2) and (3) at different places depending upon the context.

Reply:
Refactored to improve flexibility, reliability and readability.